core/src/repair/serve_repair.rs

```diff
@@ -219,7 +214,8 @@ impl RepairRequestHeader {
     }
 }
 
-pub(crate) type Ping = ping_pong::Ping<[u8; REPAIR_PING_TOKEN_SIZE]>;
+type Ping = ping_pong::Ping<REPAIR_PING_TOKEN_SIZE>;
+type PingCache = ping_pong::PingCache<REPAIR_PING_TOKEN_SIZE>;
```

Comment:
is there any reason why `REPAIR_PING_TOKEN_SIZE` may differ from `GOSSIP_PING_TOKEN_SIZE`? And would it make sense to just turn it into one variable called `PING_TOKEN_SIZE` and make it public in `gossip/src/protocol.rs`?

Reply:
```
debug_assert!(N >= std::mem::size_of::<u64>());
```
is this the only safeguard if someone accidentally defines N < 8? is it worth making sure someone can't define N < 8 for non debug builds?

Reply:
Yes, I have commented this out where `struct Ping` is defined.

> is it worth making sure someone can't define N < 8 for non debug builds?

how?

Reply:
could we just change that line to:
```
const_assert!(N >= std::mem::size_of::<u64>());
```
This would fail at compile time and wouldn't add runtime overhead.

Reply:
I tried that before, but `const_assert` wouldn't work because `N` is const generic parameter, and apparently static_assertions cannot handle it.